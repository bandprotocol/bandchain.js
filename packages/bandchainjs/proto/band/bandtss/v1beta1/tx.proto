syntax = "proto3";
package band.bandtss.v1beta1;

import "gogoproto/gogo.proto";
import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";
import "amino/amino.proto";
import "cosmos/base/v1beta1/coin.proto";
import "cosmos_proto/cosmos.proto";
import "cosmos/msg/v1/msg.proto";

import "band/bandtss/v1beta1/genesis.proto";

option go_package = "github.com/bandprotocol/chain/v3/x/bandtss/types";

// Msg defines the bandtss Msg service.
service Msg {
  option (cosmos.msg.v1.service) = true;
  // RequestSignature submits a general message to be signed by a specific group.
  rpc RequestSignature(MsgRequestSignature) returns (MsgRequestSignatureResponse);

  // Activate activates the status of the sender.
  rpc Activate(MsgActivate) returns (MsgActivateResponse);

  // UpdateParams updates the x/bandtss parameters.
  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);

  // TransitionGroup creates a request for creating a new group and replacing current group.
  rpc TransitionGroup(MsgTransitionGroup) returns (MsgTransitionGroupResponse);

  // ForceTransitionGroup sets the given group to the incoming group without the signature of a transition
  // message from a current group.
  rpc ForceTransitionGroup(MsgForceTransitionGroup) returns (MsgForceTransitionGroupResponse);
}

// MsgRequestSignature is a request message used for initiating the signing process.
message MsgRequestSignature {
  option (cosmos.msg.v1.signer)      = "sender";
  option (amino.name)                = "bandtss/MsgRequestSignature";
  option (gogoproto.goproto_getters) = false;

  // content is the signature order of this request signature message.
  google.protobuf.Any content = 1 [(cosmos_proto.accepts_interface) = "Content"];
  // memo is the additional note of the message.
  string memo = 2;
  // fee_limit is the maximum tokens that will be paid for this request.
  repeated cosmos.base.v1beta1.Coin fee_limit = 3
      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
  // sender is the requester of the signing process.
  string sender = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgRequestSignatureResponse is response data for MsgRequestSignature message
message MsgRequestSignatureResponse {}

// MsgActivate is a message used to activate the status of the sender.
message MsgActivate {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "bandtss/MsgActivate";

  // address is the signer of this message, who must be a member of the group.
  string sender = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
  // group_id is the group id of the member.
  uint64 group_id = 2
      [(gogoproto.customname) = "GroupID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"];
}

// MsgActivateResponse is response data for MsgActivate message
message MsgActivateResponse {}

// MsgUpdateParams is the Msg/UpdateParams request type.
message MsgUpdateParams {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "bandtss/MsgUpdateParams";

  // params defines the x/tss parameters to update.
  Params params = 1 [(gogoproto.nullable) = false];
  // authority is the address that controls the module (defaults to x/gov unless overwritten).
  string authority = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgUpdateParamsResponse defines the response structure for executing a MsgUpdateParams message.
message MsgUpdateParamsResponse {}

// MsgTransitionGroup is the Msg/TransitionGroup request type.
message MsgTransitionGroup {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "bandtss/MsgTransitionGroup";

  // members is a list of members in this group.
  repeated string members = 1;
  // threshold is a minimum number of members required to produce a signature.
  uint64 threshold = 2;
  // exec_time is the time that will be substituted in place of the group.
  google.protobuf.Timestamp exec_time = 3 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
  // authority is the address that controls the module (defaults to x/gov unless overwritten).
  string authority = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgTransitionGroupResponse is the Msg/TransitionGroup response type.
message MsgTransitionGroupResponse {}

// MsgForceTransitionGroup is the Msg/ForceTransitionGroup request type.
message MsgForceTransitionGroup {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "bandtss/ForceTransitionGroup";

  // incoming_group_id is the ID of the group that the module want to transition to.
  uint64 incoming_group_id = 1 [
    (gogoproto.customname) = "IncomingGroupID",
    (gogoproto.casttype)   = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"
  ];
  // exec_time is the time that will be substituted in place of the group.
  google.protobuf.Timestamp exec_time = 2 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
  // authority is the address that controls the module (defaults to x/gov unless overwritten).
  string authority = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgForceTransitionGroupResponse is the Msg/ForceTransitionGroup response type.
message MsgForceTransitionGroupResponse {}
