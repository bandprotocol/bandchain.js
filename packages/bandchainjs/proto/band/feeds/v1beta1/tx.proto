syntax = "proto3";
package band.feeds.v1beta1;

option go_package = "github.com/bandprotocol/chain/v3/x/feeds/types";

import "gogoproto/gogo.proto";
import "band/feeds/v1beta1/feeds.proto";
import "band/feeds/v1beta1/params.proto";
import "cosmos_proto/cosmos.proto";
import "cosmos/msg/v1/msg.proto";
import "amino/amino.proto";

// Msg defines the feeds Msg service.
service Msg {
  option (cosmos.msg.v1.service) = true;

  // Vote is an RPC method to vote signal ids and their powers.
  rpc Vote(MsgVote) returns (MsgVoteResponse);

  // SubmitSignalPrices is an RPC method to submit signal prices.
  rpc SubmitSignalPrices(MsgSubmitSignalPrices) returns (MsgSubmitSignalPricesResponse);

  // UpdateReferenceSourceConfig is an RPC method to update reference price source configuration.
  rpc UpdateReferenceSourceConfig(MsgUpdateReferenceSourceConfig) returns (MsgUpdateReferenceSourceConfigResponse);

  // UpdateParams is an RPC method to update parameters.
  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
}

// MsgVote is the transaction message to vote signals.
message MsgVote {
  option (cosmos.msg.v1.signer) = "voter";
  option (amino.name)           = "feeds/MsgVote";

  // voter is the address of the voter that wants to vote signals.
  string voter = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // signals is a list of submitted signals.
  repeated Signal signals = 2 [(gogoproto.nullable) = false];
}

// MsgVoteResponse is the response type for the Msg/Vote RPC method.
message MsgVoteResponse {}

// MsgSubmitSignalPrices is the transaction message to submit multiple signal prices.
message MsgSubmitSignalPrices {
  option (cosmos.msg.v1.signer) = "validator";
  option (amino.name)           = "feeds/MsgSubmitSignalPrices";

  // validator is the address of the validator that is performing the operation.
  string validator = 1 [(cosmos_proto.scalar) = "cosmos.ValidatorAddressString"];

  // timestamp is the timestamp used as reference for the data.
  int64 timestamp = 2;

  // signal_prices is a list of signal prices to submit.
  repeated SignalPrice signal_prices = 3 [(gogoproto.nullable) = false];
}

// MsgSubmitSignalPricesResponse is the response type for the Msg/SubmitSignalPrices RPC method.
message MsgSubmitSignalPricesResponse {}

// MsgUpdateReferenceSourceConfig is the transaction message to update reference price source's configuration.
message MsgUpdateReferenceSourceConfig {
  option (cosmos.msg.v1.signer) = "admin";
  option (amino.name)           = "feeds/MsgUpdateReferenceSourceConfig";

  // admin is the address of the admin that is performing the operation.
  string admin = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // reference_source_config is the information of reference price source.
  ReferenceSourceConfig reference_source_config = 2 [(gogoproto.nullable) = false];
}

// MsgUpdateReferenceSourceConfigResponse is the response type for the Msg/UpdateReferenceSourceConfig RPC method.
message MsgUpdateReferenceSourceConfigResponse {}

// MsgUpdateParams is the transaction message to update parameters.
message MsgUpdateParams {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "feeds/MsgUpdateParams";

  // authority is the address of the governance account.
  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // params is the x/feeds parameters to update.
  Params params = 2 [(gogoproto.nullable) = false];
}

// MsgUpdateParamsResponse is the response type for the Msg/UpdateParams RPC method.
message MsgUpdateParamsResponse {}
