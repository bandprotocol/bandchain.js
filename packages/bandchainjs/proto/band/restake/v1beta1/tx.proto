syntax = "proto3";
package band.restake.v1beta1;

option go_package = "github.com/bandprotocol/chain/v3/x/restake/types";

import "amino/amino.proto";
import "gogoproto/gogo.proto";
import "cosmos/msg/v1/msg.proto";
import "cosmos_proto/cosmos.proto";
import "cosmos/base/v1beta1/coin.proto";
import "band/restake/v1beta1/genesis.proto";

// Msg defines the restake Msg service.
service Msg {
  option (cosmos.msg.v1.service) = true;

  // Stake defines a method for staking coins into the module.
  rpc Stake(MsgStake) returns (MsgStakeResponse);

  // Unstake defines a method for unstaking coins out of the module.
  rpc Unstake(MsgUnstake) returns (MsgUnstakeResponse);

  // UpdateParams defines a method for updating parameters.
  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
}

// MsgStake is the request message type for staking coins.
message MsgStake {
  option (cosmos.msg.v1.signer) = "staker_address";
  option (amino.name)           = "restake/MsgStake";

  // staker_address is the address that will stake the coins.
  string staker_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // coins is the coins that will be staked.
  repeated cosmos.base.v1beta1.Coin coins = 2 [
    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
    (gogoproto.nullable)     = false,
    (amino.dont_omitempty)   = true
  ];
}

// MsgStakeResponse is the response message type for staking coins.
message MsgStakeResponse {}

// MsgUnstake is the request message type for unstaking coins.
message MsgUnstake {
  option (cosmos.msg.v1.signer) = "staker_address";
  option (amino.name)           = "restake/MsgUnstake";

  // staker_address is the address that will unstake the coins.
  string staker_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // coins is the coins that will be unstaked.
  repeated cosmos.base.v1beta1.Coin coins = 2 [
    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
    (gogoproto.nullable)     = false,
    (amino.dont_omitempty)   = true
  ];
}

// MsgUnstakeResponse is the response message type for unstaking coins.
message MsgUnstakeResponse {}

// MsgUpdateParams is the transaction message to update parameters.
message MsgUpdateParams {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "restake/MsgUpdateParams";

  // authority is the address of the governance account.
  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // params is parameters to update.
  Params params = 2 [(gogoproto.nullable) = false];
}

// MsgUpdateParamsResponse is the response type for the Msg/UpdateParams RPC method.
message MsgUpdateParamsResponse {}
