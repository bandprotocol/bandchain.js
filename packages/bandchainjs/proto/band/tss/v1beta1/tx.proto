syntax = "proto3";
package band.tss.v1beta1;

import "gogoproto/gogo.proto";
import "amino/amino.proto";
import "cosmos_proto/cosmos.proto";
import "cosmos/msg/v1/msg.proto";

import "band/tss/v1beta1/tss.proto";
import "band/tss/v1beta1/genesis.proto";

option go_package = "github.com/bandprotocol/chain/v3/x/tss/types";

// Msg defines the tss Msg service.
service Msg {
  option (cosmos.msg.v1.service) = true;
  // SubmitDKGRound1 submits dkg for computing round 1.
  rpc SubmitDKGRound1(MsgSubmitDKGRound1) returns (MsgSubmitDKGRound1Response);

  // SubmitDKGRound2 submits dkg for computing round 2.
  rpc SubmitDKGRound2(MsgSubmitDKGRound2) returns (MsgSubmitDKGRound2Response);

  // Complain submits proof for complaining malicious.
  rpc Complain(MsgComplain) returns (MsgComplainResponse);

  // Confirm submits own signature for proof that it can derive the secret.
  rpc Confirm(MsgConfirm) returns (MsgConfirmResponse);

  // SubmitDEs submits list of pre-commits DE for signing process.
  rpc SubmitDEs(MsgSubmitDEs) returns (MsgSubmitDEsResponse);

  // SubmitSignature submits signature on task participant need to do.
  rpc SubmitSignature(MsgSubmitSignature) returns (MsgSubmitSignatureResponse);

  // UpdateParams defines a governance operation for updating the x/tss module
  // parameters.
  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
}

// MsgSubmitDKGRound1 is the Msg/MsgSubmitDKGRound1 request type.
message MsgSubmitDKGRound1 {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "tss/MsgSubmitDKGRound1";

  // group_id is ID of the group.
  uint64 group_id = 1
      [(gogoproto.customname) = "GroupID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"];
  // round1_info is all data that require to handle round 1.
  Round1Info round1_info = 2 [(gogoproto.nullable) = false];
  // sender is the user address that submits the group creation information;
  // must be a member of this group.
  string sender = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgSubmitDKGRound1Response is the Msg/SubmitDKGRound1 response type.
message MsgSubmitDKGRound1Response {}

// MsgSubmitDKGRound2 is the Msg/SubmitDKGRound2 request type.
message MsgSubmitDKGRound2 {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "tss/MsgSubmitDKGRound2";

  // group_id is ID of the group.
  uint64 group_id = 1
      [(gogoproto.customname) = "GroupID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"];
  // round2_info is all data that is required to handle round 2.
  Round2Info round2_info = 2 [(gogoproto.nullable) = false];
  // sender is the user address that submits the group creation information;
  // must be a member of this group.
  string sender = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgSubmitDKGRound2Response is response data for MsgSubmitDKGRound2 message
message MsgSubmitDKGRound2Response {}

// MsgComplain is a message used to complain about malicious actors in the group.
message MsgComplain {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "tss/MsgComplaint";

  // group_id is ID of the group.
  uint64 group_id = 1
      [(gogoproto.customname) = "GroupID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"];
  // complaints is a list of complaints.
  repeated Complaint complaints = 2 [(gogoproto.nullable) = false];
  // sender is the user address that submits the group creation information;
  // must be a member of this group.
  string sender = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgComplainResponse is response data for MsgComplain message
message MsgComplainResponse {}

// MsgConfirm is a message used to confirm own public key.
message MsgConfirm {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "tss/MsgConfirm";

  // group_id is ID of the group.
  uint64 group_id = 1
      [(gogoproto.customname) = "GroupID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.GroupID"];
  // member_id is ID of the sender.
  uint64 member_id = 2
      [(gogoproto.customname) = "MemberID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.MemberID"];
  // own_pub_key_sig is a signature of the member_i on its own PubKey to confirm
  // that the address is able to derive the PubKey.
  bytes own_pub_key_sig = 3 [(gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.Signature"];
  // sender is the user address that submits the group creation information;
  // must be a member of this group.
  string sender = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgConfirmResponse is response data for MsgConfirm message
message MsgConfirmResponse {}

// MsgSubmitDEs is a message used to submit a list of DEs.
message MsgSubmitDEs {
  option (cosmos.msg.v1.signer) = "sender";
  option (amino.name)           = "tss/MsgSubmitDEs";

  // des is a list of DE objects.
  repeated DE des = 1 [(gogoproto.customname) = "DEs", (gogoproto.nullable) = false];
  // sender is the user address that submits DE objects.
  string sender = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgSubmitDEsResponse is response data for MsgSubmitDEs message
message MsgSubmitDEsResponse {}

// MsgSubmitSignature is a message used to submitting signature data.
message MsgSubmitSignature {
  option (cosmos.msg.v1.signer) = "signer";
  option (amino.name)           = "tss/MsgSubmitSignature";

  // signing_id is the unique identifier of the signing process.
  uint64 signing_id = 1 [
    (gogoproto.customname) = "SigningID",
    (gogoproto.casttype)   = "github.com/bandprotocol/chain/v3/pkg/tss.SigningID"
  ];
  // member_id is the unique identifier of the signer in the group.
  uint64 member_id = 2
      [(gogoproto.customname) = "MemberID", (gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.MemberID"];
  // signature is the signature produced by the signer.
  bytes signature = 3 [(gogoproto.casttype) = "github.com/bandprotocol/chain/v3/pkg/tss.Signature"];
  // signer is the address who signs a message; must be a member of the group.
  string signer = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgSubmitSignatureResponse is response data for MsgSubmitSignature message
message MsgSubmitSignatureResponse {}

// MsgUpdateParams is the Msg/UpdateParams request type.
message MsgUpdateParams {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "tss/MsgUpdateParams";

  // params defines the x/tss parameters to update.
  Params params = 1 [(gogoproto.nullable) = false];
  // authority is the address of the governance account.
  string authority = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgUpdateParamsResponse defines the response structure for executing a
// MsgUpdateParams message.
message MsgUpdateParamsResponse {}
