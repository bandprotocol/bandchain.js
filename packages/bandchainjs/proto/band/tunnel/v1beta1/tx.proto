syntax = "proto3";
package band.tunnel.v1beta1;

option go_package = "github.com/bandprotocol/chain/v3/x/tunnel/types";

import "google/protobuf/any.proto";
import "gogoproto/gogo.proto";
import "cosmos_proto/cosmos.proto";
import "cosmos/base/v1beta1/coin.proto";
import "cosmos/msg/v1/msg.proto";
import "amino/amino.proto";

import "band/tunnel/v1beta1/params.proto";
import "band/tunnel/v1beta1/tunnel.proto";

// Service definition for Msg.
service Msg {
  option (cosmos.msg.v1.service) = true;

  // CreateTunnel is a RPC method to create a new tunnel.
  rpc CreateTunnel(MsgCreateTunnel) returns (MsgCreateTunnelResponse);

  // UpdateAndResetTunnel is a RPC method to update a tunnel information and reset the interval.
  rpc UpdateAndResetTunnel(MsgUpdateAndResetTunnel) returns (MsgUpdateAndResetTunnelResponse);

  // Activate is a RPC method to activate a tunnel.
  rpc Activate(MsgActivate) returns (MsgActivateResponse);

  // Deactivate is a RPC method to deactivate a tunnel.
  rpc Deactivate(MsgDeactivate) returns (MsgDeactivateResponse);

  // TriggerTunnel is a RPC method to manually trigger a tunnel.
  rpc TriggerTunnel(MsgTriggerTunnel) returns (MsgTriggerTunnelResponse);

  // DepositToTunnel is a RPC method to deposit to an existing tunnel.
  rpc DepositToTunnel(MsgDepositToTunnel) returns (MsgDepositToTunnelResponse);

  // WithdrawFromTunnel is a RPC method to withdraw a deposit from an existing tunnel.
  rpc WithdrawFromTunnel(MsgWithdrawFromTunnel) returns (MsgWithdrawFromTunnelResponse);

  // UpdateParams is a RPC method to update parameters
  rpc UpdateParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
}

// MsgCreateTunnel is the transaction message to create a new tunnel.
message MsgCreateTunnel {
  option (cosmos.msg.v1.signer) = "creator";
  option (amino.name)           = "tunnel/MsgCreateTunnel";

  // signal_deviations is the list of signal deviations.
  repeated SignalDeviation signal_deviations = 1 [(gogoproto.nullable) = false];
  // interval is the interval for delivering the signal prices in seconds.
  uint64 interval = 2;
  // route is the route for delivering the signal prices
  google.protobuf.Any route = 3 [(cosmos_proto.accepts_interface) = "RouteI"];
  // initial_deposit is the deposit value that must be paid at tunnel creation.
  repeated cosmos.base.v1beta1.Coin initial_deposit = 4 [
    (gogoproto.nullable)     = false,
    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
    (amino.dont_omitempty)   = true
  ];
  // creator is the address of the creator.
  string creator = 5 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgCreateTunnelResponse is the response type for the Msg/CreateTunnel RPC method.
message MsgCreateTunnelResponse {
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
}

// MsgUpdateAndResetTunnel is the transaction message to update a tunnel information
// and reset the interval.
message MsgUpdateAndResetTunnel {
  option (cosmos.msg.v1.signer) = "creator";
  option (amino.name)           = "tunnel/MsgUpdateAndResetTunnel";

  // tunnel_id is the ID of the tunnel to edit.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
  // signal_deviations is the list of signal deviations.
  repeated SignalDeviation signal_deviations = 2 [(gogoproto.nullable) = false];
  // interval is the interval for delivering the signal prices.
  uint64 interval = 3;
  // creator is the address of the creator.
  string creator = 4 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgUpdateAndResetTunnelResponse is the response type for the Msg/UpdateAndResetTunnel RPC method.
message MsgUpdateAndResetTunnelResponse {}

// Activate is the transaction message to activate a tunnel.
message MsgActivate {
  option (cosmos.msg.v1.signer) = "creator";
  option (amino.name)           = "tunnel/MsgActivate";

  // tunnel_id is the ID of the tunnel to activate.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
  // creator is the address of the creator.
  string creator = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgActivateResponse is the response type for the Msg/Activate RPC method.
message MsgActivateResponse {}

// MsgDeactivate is the transaction message to deactivate a tunnel.
message MsgDeactivate {
  option (cosmos.msg.v1.signer) = "creator";
  option (amino.name)           = "tunnel/MsgDeactivate";

  // tunnel_id is the ID of the tunnel to deactivate.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
  // creator is the address of the creator.
  string creator = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgDeactivateTunnelResponse is the response type for the Msg/Deactivate RPC method.
message MsgDeactivateResponse {}

// MsgTriggerTunnel is the transaction message to manually trigger a tunnel.
message MsgTriggerTunnel {
  option (cosmos.msg.v1.signer) = "creator";
  option (amino.name)           = "tunnel/MsgTriggerTunnel";

  // tunnel_id is the ID of the tunnel to manually trigger.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];
  // creator is the address of the creator.
  string creator = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgTriggerTunnelResponse is the response type for the Msg/TriggerTunnel RPC method.
message MsgTriggerTunnelResponse {}

// MsgDepositToTunnel defines a message to deposit to an existing tunnel.
message MsgDepositToTunnel {
  option (cosmos.msg.v1.signer) = "depositor";
  option (amino.name)           = "tunnel/MsgDepositToTunnel";

  // tunnel_id defines the unique id of the tunnel.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];

  // amount to be deposited by depositor.
  repeated cosmos.base.v1beta1.Coin amount = 2 [
    (gogoproto.nullable)     = false,
    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
    (amino.dont_omitempty)   = true
  ];

  // depositor defines the deposit addresses from the tunnel.
  string depositor = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgDepositToTunnelResponse defines the Msg/DepositToTunnel response type.
message MsgDepositToTunnelResponse {}

// MsgWithdrawFromTunnel is the transaction message to withdraw a deposit from an existing tunnel.
message MsgWithdrawFromTunnel {
  option (cosmos.msg.v1.signer) = "withdrawer";
  option (amino.name)           = "tunnel/MsgWithdrawFromTunnel";

  // tunnel_id defines the unique id of the tunnel.
  uint64 tunnel_id = 1 [(gogoproto.customname) = "TunnelID"];

  // amount to be withdrawn by withdrawer.
  repeated cosmos.base.v1beta1.Coin amount = 2 [
    (gogoproto.nullable)     = false,
    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
    (amino.dont_omitempty)   = true
  ];

  // withdrawer defines the withdraw addresses from the tunnel.
  string withdrawer = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
}

// MsgWithdrawFromTunnelResponse defines the Msg/WithdrawFromTunnel response type.
message MsgWithdrawFromTunnelResponse {}

// MsgUpdateParams is the transaction message to update parameters.
message MsgUpdateParams {
  option (cosmos.msg.v1.signer) = "authority";
  option (amino.name)           = "tunnel/MsgUpdateParams";

  // authority is the address of the governance account.
  string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];

  // params is the x/tunnel parameters to update.
  Params params = 2 [(gogoproto.nullable) = false];
}

// MsgUpdateParamsResponse is the response type for the Msg/UpdateParams RPC method.
message MsgUpdateParamsResponse {}
