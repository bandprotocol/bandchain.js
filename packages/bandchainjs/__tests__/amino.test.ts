import { coin } from "@cosmjs/amino";

import { band } from "../src";
import { AminoConverter } from "../src/codegen/band/oracle/v1/tx.amino";

it("AminoConverter", async () => {
  const msg = AminoConverter["/band.oracle.v1.MsgRequestData"].toAmino({
    oracleScriptId: BigInt(1),
    calldata: new Uint8Array([1, 2, 3]),
    askCount: BigInt(1),
    minCount: BigInt(1),
    clientId: "client",
    feeLimit: [coin("1", "uband")],
    prepareGas: BigInt(1),
    executeGas: BigInt(1),
    sender: "sender",
    tssEncoder: 0,
  });
  expect(msg).toMatchSnapshot();

  const back = AminoConverter["/band.oracle.v1.MsgRequestData"].fromAmino(msg);

  expect(back).toMatchSnapshot();

  const final = AminoConverter["/band.oracle.v1.MsgRequestData"].toAmino(back);

  expect(final).toMatchSnapshot();

  expect(final).toEqual(msg);
});

it("AminoConverter.scoped", async () => {
  const msg = band.oracle.v1.AminoConverter[
    "/band.oracle.v1.MsgRequestData"
  ].toAmino({
    oracleScriptId: BigInt(1),
    calldata: new Uint8Array([1, 2, 3]),
    askCount: BigInt(1),
    minCount: BigInt(1),
    clientId: "client",
    feeLimit: [coin("1", "uband")],
    prepareGas: BigInt(1),
    executeGas: BigInt(1),
    sender: "sender",
    tssEncoder: 0,
  });
  expect(msg).toMatchSnapshot();

  const back =
    band.oracle.v1.AminoConverter["/band.oracle.v1.MsgRequestData"].fromAmino(
      msg
    );

  expect(back).toMatchSnapshot();

  const final =
    band.oracle.v1.AminoConverter["/band.oracle.v1.MsgRequestData"].toAmino(
      back
    );

  expect(final).toMatchSnapshot();

  expect(final).toEqual(msg);
});
