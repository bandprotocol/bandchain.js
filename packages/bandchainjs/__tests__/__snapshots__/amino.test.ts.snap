// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AminoConverter 1`] = `
{
  "ask_count": "1",
  "calldata": "AQID",
  "client_id": "client",
  "execute_gas": "1",
  "fee_limit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "min_count": "1",
  "oracle_script_id": "1",
  "prepare_gas": "1",
  "sender": "sender",
}
`;

exports[`AminoConverter 2`] = `
{
  "askCount": 1n,
  "calldata": Uint8Array [
    1,
    2,
    3,
  ],
  "clientId": "client",
  "executeGas": 1n,
  "feeLimit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "minCount": 1n,
  "oracleScriptId": 1n,
  "prepareGas": 1n,
  "sender": "sender",
}
`;

exports[`AminoConverter 3`] = `
{
  "ask_count": "1",
  "calldata": "AQID",
  "client_id": "client",
  "execute_gas": "1",
  "fee_limit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "min_count": "1",
  "oracle_script_id": "1",
  "prepare_gas": "1",
  "sender": "sender",
}
`;

exports[`AminoConverter.scoped 1`] = `
{
  "ask_count": "1",
  "calldata": "AQID",
  "client_id": "client",
  "execute_gas": "1",
  "fee_limit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "min_count": "1",
  "oracle_script_id": "1",
  "prepare_gas": "1",
  "sender": "sender",
}
`;

exports[`AminoConverter.scoped 2`] = `
{
  "askCount": 1n,
  "calldata": Uint8Array [
    1,
    2,
    3,
  ],
  "clientId": "client",
  "executeGas": 1n,
  "feeLimit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "minCount": 1n,
  "oracleScriptId": 1n,
  "prepareGas": 1n,
  "sender": "sender",
}
`;

exports[`AminoConverter.scoped 3`] = `
{
  "ask_count": "1",
  "calldata": "AQID",
  "client_id": "client",
  "execute_gas": "1",
  "fee_limit": [
    {
      "amount": "1",
      "denom": "uband",
    },
  ],
  "min_count": "1",
  "oracle_script_id": "1",
  "prepare_gas": "1",
  "sender": "sender",
}
`;
